### ISSUE TITLE
Completely document all of ZUN's original file formats

### ISSUE THREAD

**Issue by nmlgc** [thpatch/thpatch#3]
Currently, the documentation we _have_ on ZUN's original formats is pretty fragmented. It started with xarnonymous adding documentation to the [original Touhou Toolkit wiki](https://code.google.com/p/thtk/w/). Then, [Link Mauve documented some more parts for PyTouhou](http://pytouhou.linkmauve.fr/doc/), with more being effectively inside the PyTouhou sources and the [patch queue](http://pytouhou.linkmauve.fr/patches/), @Nutzer [documented some other parts for his modding projects](https://github.com/Nutzer/Insfinder), together with [a German tutorial of ECL as a whole](
http://nutzer.bplaced.net/Database/tut/), and I too have some random documentation lying around in my old development notes somewhere, from back when this project started.

I think it's time to finally have a common place for this sort of thing, especially since you're going to take documentation as a whole more seriously on the new site. Updating that section should then be an integral part of patch development for future games.

This issue could maybe be moved to [the Touhou Toolkit repository](/thpatch/thtk) if you prefer man pages over websites.

**Comment by nmlgc** [comment]
Two more ECL reference pages cropped up recently: [A new one by @Nutzer](http://www.nutzer.bplaced.net/Database/), and [a Google spreadsheet by Dark Kitsune](https://www.shrinemaiden.org/forum/index.php?topic=18846.0).

**Comment by tirr-c** [comment]
When ECL reference is complete, should ins__xxx_'s of thecl be replaced by appropriate instruction names(opcodes)?

**Comment by linkmauve** [comment]
Yeah, I already had a patch for that long ago, but the bison part wasn’t working properly so I never published it.
But if you work with ECL long enough, you’ll end up reading numeric instructions as fast as names anyway. :D

**Comment by nmlgc** [comment]
[Here's another reference of TH14's ECL format.](https://thwiki.cc/%E8%84%9A%E6%9C%AC%E5%AF%B9%E7%85%A7%E8%A1%A8/ECL)

**Comment by tirr-c** [comment]
Oh, it needs translation.

**Comment by nmlgc** [comment]
[More documentation](https://en.touhouwiki.net/wiki/User:Mddass/Touhou_File_Format_Specification) by @ManDude.

**Comment by nmlgc** [comment]
[.sht documentation for ≥TH10, together with a file editor](https://github.com/Priw8/sht-webedit), by @Priw8.
